For some reason, when building with --without-stlport4 on Linux/gcc 3.4.4,
::std::swap triggers

bits/stl_algobase.h: In function `void std::swap(_Tp&, _Tp&) [with _Tp = std::auto_ptr<SvxEditSource>]':
svx/source/accessibility/AccessibleEmptyEditSource.cxx:320:   instantiated from here
bits/stl_algobase.h:132: error: passing `const std::auto_ptr<SvxEditSource>' as `this' argument of `std::auto_ptr<_Tp>::operator std::auto_ptr_ref<_Tp1>() [with _Tp1 = SvxEditSource, _Tp = SvxEditSource]' discards qualifiers

Better fix (or explanation what exactly causes std::swap to cast this around)
welcome, but for now this one works

--- svx/source/accessibility/AccessibleEmptyEditSource.cxx.ark	2005-08-06 10:00:38.000000000 +0000
+++ svx/source/accessibility/AccessibleEmptyEditSource.cxx	2005-08-06 10:34:51.000000000 +0000
@@ -317,7 +317,9 @@
             EndListening( *mrObj.GetModel() );
 
         ::std::auto_ptr< SvxEditSource > pProxySource( new AccessibleProxyEditSource_Impl(mrObj, mrView, mrViewWindow) );
-        ::std::swap(pProxySource, mpEditSource);
+	::std::auto_ptr< SvxEditSource > tmp = mpEditSource;
+	mpEditSource = pProxySource;
+	pProxySource = tmp;
         
         // register as listener
         StartListening( mpEditSource->GetBroadcaster() );
